### PR TITLE
Support more encodings for Kafka in OTel Ingester

### LINKED ISSUE #2579: Jaeger Opentelemetry Ingester cannot correctly handle `kafka.consumer.encoding` flag
**Describe the bug**
Jaeger Opentelemetry Ingester cannot correctly handle `kafka.consumer.encoding` when its value is `zipkin-thrift`. Also, the flag not accept the `otlp-proto` as value even Ingester's Kafka implementation is from OTel.

**To Reproduce**
1. Check helper
```bash
$ opentelemetry-ingester -h

...
--kafka.consumer.encoding string                     The encoding of spans ("json", "protobuf", "zipkin-thrift") consumed from kafka (default "protobuf")
...
```

2. Use `zipkin-thrift` as encoding value

```bash
$ opentelemetry-ingester --kafka.consumer.encoding zipkin-thrift
2020-10-20T14:04:31.624+0800	INFO	service/service.go:396	Starting Jaeger OpenTelemetry Ingester...	{"Version": "v1.20.0", "GitHash": "bd59f13478cdd35b41e85f67a6f04b6cb4d92df8", "NumCPU": 4}
2020-10-20T14:04:31.625+0800	INFO	service/service.go:240	Setting up own telemetry...
2020-10-20T14:04:31.629+0800	INFO	service/telemetry.go:110	Serving Prometheus metrics	{"address": "localhost:8888", "legacy_metrics": false, "new_metrics": true, "level": 0, "service.instance.id": "28b6a4ce-d467-4a53-8941-06840458019e"}
panic: zipkin-thrift is not a supported kafka encoding in the OTEL collector.

goroutine 1 [running]:
github.com/jaegertracing/jaeger/cmd/opentelemetry/app/receiver/kafkareceiver.MustOtelEncodingForJaegerEncoding(...)
	github.com/jaegertracing/jaeger/cmd/opentelemetry/app/receiver/kafkareceiver/kafka_receiver.go:158
github.com/jaegertracing/jaeger/cmd/opentelemetry/app/receiver/kafkareceiver.(*Factory).CreateDefaultConfig(0xc000488140, 0x46774a0, 0xc0008ee4b0)
	github.com/jaegertracing/jaeger/cmd/opentelemetry/app/receiver/kafkareceiver/kafka_receiver.go:62 +0x7e8
go.opentelemetry.io/collector/config/configcheck.ValidateConfigFromFactories(0xc0003de840, 0xc0003de900, 0xc0003de8d0, 0xc0003de7e0, 0x0, 0xc000407560)
	go.opentelemetry.io/collector@v0.12.0/config/configcheck/configcheck.go:41 +0xe2
go.opentelemetry.io/collector/service.(*Application).setupConfigurationComponents(0xc000846000, 0x468dba0, 0xc0000560c8, 0xc000409610, 0x0, 0x3)
	go.opentelemetry.io/collector@v0.12.0/service/service.go:273 +0x5a
go.opentelemetry.io/collector/service.(*Application).execute(0xc000846000, 0x468dba0, 0xc0000560c8, 0xc000409610, 0x0, 0x0)
	go.opentelemetry.io/collector@v0.12.0/service/service.go:414 +0x4e5
go.opentelemetry.io/collector/service.New.func1(0xc00004e580, 0xc000489780, 0x0, 0x2, 0x0, 0x0)
	go.opentelemetry.io/collector@v0.12.0/service/service.go:154 +0xa5
github.com/spf13/cobra.(*Command).execute(0xc00004e580, 0xc00004c1c0, 0x2, 0x2, 0xc00004e580, 0xc00004c1c0)
	github.com/spf13/cobra@v1.0.0/command.go:842 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0xc00004e580, 0x0, 0x0, 0x3)
	github.com/spf13/cobra@v1.0.0/command.go:950 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.0.0/command.go:887
go.opentelemetry.io/collector/service.(*Application).Run(...)
	go.opentelemetry.io/collector@v0.12.0/service/service.go:467
main.main()
	command-line-arguments/main.go:97 +0x8af
```

**Expected behavior**
Accept `zipkin-thrift` or `otlp-proto`as its value since these values (nearly) work in the config file.

**Version (please complete the following information):**
 - OS: macOS 10.15.6
 - Jaeger version: 3d998711700382c6729b83fa66f8b20a3aae5116 (master)

### PR DESCRIPTION
**Pull request by XSAM**
Resolves #2579

### CODE REVIEW

File: cmd/ingester/app/flags.go [hunk 1]
```diff
@@ -114,7 +114,12 @@ func AddOTELFlags(flagSet *flag.FlagSet) {
 	flagSet.String(
 		KafkaConsumerConfigPrefix+SuffixEncoding,
 		DefaultEncoding,
-		fmt.Sprintf(`The encoding of spans ("%s") consumed from kafka`, strings.Join(kafka.AllEncodings, "\", \"")))
+		fmt.Sprintf(`The encoding of spans ("%s") consumed from kafka`, strings.Join(
+			append(kafka.AllEncodings,
+				kafka.EncodingZipkinJSON,
```
Comment (objectiser) [inline cmd/ingester/app/flags.go:119]: I think these options would need to be qualified to indicate only supported with the otel image.

File: plugin/storage/kafka/options.go [hunk 2]
```diff
@@ -34,6 +34,12 @@ const (
 	EncodingProto = "protobuf"
 	// EncodingZipkinThrift is used for spans encoded as Zipkin Thrift.
 	EncodingZipkinThrift = "zipkin-thrift"
+	// EncodingZipkinProto is used for spans encoded as Zipkin Protobuf.
```
Comment (objectiser) [inline plugin/storage/kafka/options.go:37]: Same with these comments - can you make it clear they are only relevant for otel images.

File: cmd/opentelemetry/app/receiver/kafkareceiver/kafka_receiver.go [hunk 3]
```diff
@@ -145,6 +145,14 @@ func MustOtelEncodingForJaegerEncoding(jaegerEncoding string) string {
 		return "jaeger_proto"
 	case kafka.EncodingJSON:
 		return "jaeger_json"
+	case kafka.EncodingOTLPProto:
```
Comment (joe-elliott) [inline cmd/opentelemetry/app/receiver/kafkareceiver/kafka_receiver.go:148]: This same method is used for the exporter which doesn't support some of these encodings.

**Review by joe-elliott — CHANGES_REQUESTED** [review]
I don't mind merging this PR (particularly if it allows @XSAM to do something interesting with the Jaeger OTel Ingester), but all of this functionality is still being discussed and this may change in the final version.

File: cmd/opentelemetry/app/exporter/kafkaexporter/kafka_exporter.go [hunk 4]
```diff
@@ -127,3 +126,15 @@ func (f Factory) CreateLogsExporter(
 ) (component.LogsExporter, error) {
 	return f.Wrapped.CreateLogsExporter(ctx, params, cfg)
 }
+
+// mustOtelEncodingForJaegerEncoding translates a jaeger encoding to a otel encoding
+func mustOtelEncodingForJaegerEncoding(jaegerEncoding string) string {
+	switch jaegerEncoding {
+	case kafka.EncodingProto:
```
Comment (joe-elliott) [inline cmd/opentelemetry/app/exporter/kafkaexporter/kafka_exporter.go:133]: no otlp proto?

Comment (XSAM) [reply cmd/opentelemetry/app/exporter/kafkaexporter/kafka_exporter.go:133]: My original thought is to add `otlp-proto` just for Kafka receiver, not Kafka exporter. But I'm glad to make Kafka exporter support that.

**Review by joe-elliott — DISMISSED** [review]
So I generally expect this to change, but don't think there's any issue adding this flexibility to the OTel components for now. 

@XSAM looks like you need to sign some commits?  Other than that LGTM

**Review by joe-elliott — APPROVED** [review]
Thanks for the contribution!

**Comment by yurishkuro** [comment]
Are there any documentation changes required to accompany this?

**Comment by joe-elliott** [comment]
> Are there any documentation changes required to accompany this?

Given that this is all likely to change and none of this is documented I think this is fine.  I tried to make it clear that this is all currently up for debate and may change.  I think @XSAM is just experimenting with the Otel pieces and this was something they needed.

**Pull request merged**